src/ppx_deriving.mli

```diff
@@ -42,6 +42,13 @@ type deriver = {
 (** [register deriver] registers [deriver] according to its [name] field. *)
 val register : deriver -> unit
 
+(** [add_register_hook hook] adds [hook] to be executed whenever a new deriver
+    is registered *)
+val add_register_hook : (deriver -> unit) -> unit
+
+(** [derivers ()] returns all currently registered derivers *)
```

Comment:
Nitpick: the sentences in the docs are finished with dots. I'll merge after this is fixed!